File: android-compose/app/src/main/java/com/demo/model/Device.kt
package com.demo.model

/**
 * AI wearable device model.
 */
data class Device(
    val id: String,
    val name: String,
    val battery: Int,
    val firmware: String,
    val state: DeviceState
)

enum class DeviceState {
    IDLE,
    SCANNING,
    CONNECTING,
    CONNECTED,
    ERROR
}

File: ios-swiftui/Features/Device/Device.swift
import Foundation

/// AI穿戴设备模型
struct Device: Identifiable {
    let id: String
    let name: String
    let battery: Int
    let firmware: String
    var state: DeviceState
}

/// 设备连接状态
enum DeviceState {
    case idle
    case scanning
    case connecting
    case connected
    case error(String)
}
